src/legacy/ui/public/vis/editors/default/default.js

```diff
@@ -1,200 +0,0 @@
-            if ($scope.vis.type.editorConfig.defaultSize === DefaultEditorSize.SMALL) {
-              return 'visEditor__collapsibleSidebar--small';
-            } else if ($scope.vis.type.editorConfig.defaultSize === DefaultEditorSize.MEDIUM) {
-              return 'visEditor__collapsibleSidebar--medium';
-            } else if ($scope.vis.type.editorConfig.defaultSize === DefaultEditorSize.LARGE) {
-              return 'visEditor__collapsibleSidebar--large';
-            }
-          };
-
-          $scope.$watch(() => {
-            return $scope.vis.getSerializableState($scope.state);
-          }, function (newState) {
-            $scope.vis.dirty = !angular.equals(newState, $scope.oldState);
-            const responseAggs = $scope.state.aggs.getResponseAggs();
-            $scope.hasHistogramAgg = responseAggs.some(agg => agg.type.name === 'histogram');
-            $scope.metricAggs = responseAggs.filter(agg =>
-              _.get(agg, 'schema.group') === AggGroupNames.Metrics);
-            const lastParentPipelineAgg = _.findLast($scope.metricAggs, ({ type }) => type.subtype === parentPipelineAggHelper.subtype);
-            $scope.lastParentPipelineAggTitle = lastParentPipelineAgg && lastParentPipelineAgg.type.title;
-          }, true);
-
-          // fires when visualization state changes, and we need to copy changes to editorState
-          $scope.$watch(() => {
-            return $scope.vis.getCurrentState(false);
-          }, (newState) => {
-            if (!_.isEqual(newState, $scope.oldState)) {
-              $scope.state = $scope.vis.copyCurrentState(true);
-              $scope.oldState = newState;
-            }
-          }, true);
-
-          // Load the default editor template, attach it to the DOM and compile it.
-          // It should be added to the DOM before compiling, to prevent some resize
-          // listener issues.
-          const template = $(defaultEditorTemplate);
-          this.el.html(template);
-          $compile(template)($scope);
-        } else {
-          $scope = this.$scope;
-          updateScope();
-        }
-
-        if (!this._handler) {
-          const visualizationEl = this.el.find('.visEditor__canvas')[0];
-
-          this._handler = await embeddables.getEmbeddableFactory('visualization').createFromObject(this.savedObj, {
-            uiState: uiState,
-            appState: getAppState(),
-            timeRange: timeRange,
-            filters: filters || [],
-            query: query,
-          });
-          this._handler.render(visualizationEl);
-
-        } else {
-          this._handler.updateInput({
-            timeRange: timeRange,
-            filters: filters || [],
-            query: query,
-          });
-        }
-
-      });
-    }
-
-    resize() {
-
-    }
-
-    destroy() {
-      if (this.$scope) {
-        this.$scope.$destroy();
-        this.$scope = null;
-      }
-      if (this._handler) {
-        this._handler.destroy();
```

Comment:
I guess we should destroy the handler in `default_editor.tsx` as well.

Reply:
Good catch, done!